qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,438 @@
+                - If the number of different repetition series is not three.
+        """
+        # TODO this is temporary logic.
+        self.set_analysis_options(
+            data_processor=get_to_signal_processor(
+                meas_level=self.run_options.meas_level,
+                meas_return=self.run_options.meas_return,
+            ),
+        )
+
+        xm = self.experiment_options.xm
+        xp = self.experiment_options.xp
+
+        if xp is None:
+            beta = Parameter("β")
+            with pulse.build(backend=backend, name="xp") as xp:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+            with pulse.build(backend=backend, name="xm") as xm:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=-self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+        # Now check that we are dealing with Drag pulses.
+        for schedule in [xp, xm]:
+            for block in schedule.blocks:
+                if isinstance(block, pulse.Play):
+                    if isinstance(block.pulse, pulse.Drag):
+                        break
+            else:
+                raise CalibrationError(f"No Drag pulse found in {schedule.name}.")
+
+        beta_xp = next(iter(xp.parameters))
+        beta_xm = next(iter(xm.parameters))
+
+        if beta_xp != beta_xm:
+            raise CalibrationError(
+                f"Beta for xp and xm in {self.__class__.__name__} calibration are not identical."
+            )
+
+        xp_gate = Gate(name="xp", num_qubits=1, params=[beta_xp])
+        xm_gate = Gate(name="xm", num_qubits=1, params=[beta_xp])
+
+        reps = self.experiment_options.reps
+        if len(reps) != 3:
+            raise CalibrationError(
+                f"The number of repetitions for {self.__class__.__name__} must be three. "
+                "This constraint can be removed once CurveFitting supports a dynamic number "
+                "of series."
+            )
+
+        circuits = []
+        for beta in self.experiment_options.betas:
+
+            beta = np.round(beta, decimals=6)
+
+            for idx, rep in enumerate(reps):
+                circuit = QuantumCircuit(1)
+                for index in range(rep):
+                    circuit.append(xp_gate, (0,))
+                    circuit.barrier()
```

Comment:
Is this barrier necessary? Custom gates are not merged because they don't have matrix definition. This will much simplify the code, i.e. just use `for _ in range(rep):`

Reply:
We do not need it. Changed here: 1063ca2